test/core/debug/stats_test.cc

```diff
@@ -31,138 +31,66 @@
+    return now->Diff(*begin_);
   }
 
  private:
-  grpc_stats_data begin_;
+  std::unique_ptr<GlobalStats> begin_ = global_stats().Collect();
 };
 
-TEST(StatsTest, IncCounters) {
-  for (int i = 0; i < GRPC_STATS_COUNTER_COUNT; i++) {
-    std::unique_ptr<Snapshot> snapshot(new Snapshot);
-
-    grpc_core::ExecCtx exec_ctx;
-    GRPC_STATS_INC_COUNTER((grpc_stats_counters)i);
-
-    EXPECT_EQ(snapshot->delta().counters[i], 1);
-  }
-}
-
 TEST(StatsTest, IncSpecificCounter) {
   std::unique_ptr<Snapshot> snapshot(new Snapshot);
 
-  grpc_core::ExecCtx exec_ctx;
-  GRPC_STATS_INC_CLIENT_CALLS_CREATED();
+  ExecCtx exec_ctx;
+  global_stats().IncrementClientCallsCreated();
 
-  EXPECT_EQ(snapshot->delta().counters[GRPC_STATS_COUNTER_CLIENT_CALLS_CREATED],
-            1);
+  EXPECT_EQ(snapshot->delta()->client_calls_created, 1);
 }
 
-static int FindExpectedBucket(int i, int j) {
-  if (j < 0) {
+static int FindExpectedBucket(const HistogramView& h, int value) {
+  if (value < 0) {
     return 0;
   }
-  if (j >= grpc_stats_histo_bucket_boundaries[i][grpc_stats_histo_buckets[i]]) {
-    return grpc_stats_histo_buckets[i] - 1;
+  if (value >= h.bucket_boundaries[h.num_buckets]) {
+    return h.num_buckets - 1;
   }
-  return std::upper_bound(grpc_stats_histo_bucket_boundaries[i],
-                          grpc_stats_histo_bucket_boundaries[i] +
-                              grpc_stats_histo_buckets[i],
-                          j) -
-         grpc_stats_histo_bucket_boundaries[i] - 1;
+  return std::upper_bound(h.bucket_boundaries,
+                          h.bucket_boundaries + h.num_buckets, value) -
+         h.bucket_boundaries - 1;
 }
 
 class HistogramTest : public ::testing::TestWithParam<int> {};
 
 TEST_P(HistogramTest, CheckBucket) {
-  const int kHistogram = GetParam();
-  int max_bucket_boundary =
-      grpc_stats_histo_bucket_boundaries[kHistogram]
-                                        [grpc_stats_histo_buckets[kHistogram] -
-                                         1];
+  const GlobalStats::Histogram kHistogram =
+      static_cast<GlobalStats::Histogram>(GetParam());
+  auto some_stats = absl::make_unique<GlobalStats>();
+  auto view = some_stats->histogram(kHistogram);
+  const int max_bucket_boundary = view.bucket_boundaries[view.num_buckets];
   for (int i = -1000; i < max_bucket_boundary + 1000; i++) {
-    ASSERT_EQ(FindExpectedBucket(kHistogram, i),
-              grpc_stats_get_bucket[kHistogram](i))
+    ASSERT_EQ(FindExpectedBucket(view, i), view.bucket_for(i))
         << "i=" << i << " expect_bucket="
-        << grpc_stats_histo_bucket_boundaries[kHistogram]
-                                             [FindExpectedBucket(kHistogram, i)]
-        << " actual_bucket="
-        << grpc_stats_histo_bucket_boundaries[kHistogram]
-                                             [grpc_stats_get_bucket[kHistogram](
-                                                 i)];
-  }
-}
-
-TEST_P(HistogramTest, IncHistogram) {
```

Comment:
why is this test being deleted?

Reply:
it relies on some facilities that we're losing with the new api... originally it tested two things:
1. whether we're landing in the right bucket for each value, but CheckBucket above does a much better and more efficient test for that
2. whether we can increment histograms concurrently, but a run of any test in the suite under tsan also verifies that

so instead of adding api for a test we don't need i'm suggesting drop the test!